poc_mobile_verifier/src/owner_shares.rs

```diff
@@ -1,94 +1,170 @@
-        });
-
-        Ok(Self {
-            epoch: epoch.clone(),
-            rewards,
-        })
     }
 
-    pub async fn write(
+    pub fn into_radio_shares(
         self,
-        subnet_rewards_tx: &file_sink::MessageSender,
-    ) -> file_store::Result<oneshot::Receiver<file_store::Result>> {
-        file_sink_write!(
-            "subnet_rewards",
-            subnet_rewards_tx,
-            proto::SubnetworkRewards {
-                start_epoch: self.epoch.start.timestamp() as u64,
-                end_epoch: self.epoch.end.timestamp() as u64,
-                rewards: self.rewards,
-            }
+        epoch: &'_ Range<DateTime<Utc>>,
+    ) -> impl Iterator<Item = proto::RadioRewardShare> + '_ {
+        let total_shares = self.total_shares();
+        let total_rewards = get_scheduled_tokens(epoch.start, epoch.end - epoch.start).unwrap();
+        let rewards_per_share = (total_rewards / total_shares)
+            .round_dp_with_strategy(REWARDS_PER_SHARE_PREC, RoundingStrategy::ToPositiveInfinity);
+        self.shares
+            .into_iter()
+            .flat_map(move |(owner_key, radio_shares)| {
+                radio_shares
+                    .into_iter()
+                    .map(move |radio_share| proto::RadioRewardShare {
+                        owner_key: owner_key.to_vec(),
+                        hotspot_key: radio_share.hotspot_key.to_vec(),
+                        cbsd_id: radio_share.cbsd_id,
+                        amount: {
+                            let rewards = rewards_per_share * radio_share.amount;
+                            let rewards = (rewards * Decimal::from(MOBILE_SCALE))
+                                .round_dp_with_strategy(0, RoundingStrategy::MidpointAwayFromZero);
+                            rewards.to_u64().unwrap_or(0)
+                        },
+                        start_epoch: epoch.start.encode_timestamp(),
+                        end_epoch: epoch.end.encode_timestamp(),
+                    })
+            })
+    }
+}
+
+// 100M genesis rewards per day
+const GENESIS_REWARDS_PER_DAY: i64 = 100_000_000;
+
+lazy_static! {
+    static ref GENESIS_START: DateTime<Utc> = Utc.ymd(2022, 7, 11).and_hms(0, 0, 0);
+}
+
+pub fn get_scheduled_tokens(start: DateTime<Utc>, duration: Duration) -> Option<Decimal> {
+    if *GENESIS_START <= start {
+        // Get tokens from start - duration
+        Some(
+            (Decimal::from(GENESIS_REWARDS_PER_DAY)
+                / Decimal::from(Duration::hours(24).num_seconds()))
+                * Decimal::from(duration.num_seconds()),
         )
-        .await
+    } else {
+        None
+    }
+}
+
+#[derive(thiserror::Error, Debug)]
+#[error(transparent)]
+pub struct ResolveError(#[from] tonic::Status);
+
+#[async_trait::async_trait]
+pub trait OwnerResolver: Send {
+    async fn resolve_owner(
+        &mut self,
+        address: &PublicKey,
+    ) -> Result<Option<PublicKey>, ResolveError>;
+
+    async fn owner_shares(
```

Comment:
i'm not sure i like hanging this off the OwnerResolver impl b/c in practice it means we're calling it against the follower client which just reads as confusing; it implies the follower is calculating the owner shares. since this is just calling the `resolve_owner` method of the impl this feels like it should just be an independent function with the impl of the `OwnerResolver` as one of the parameters

Reply:
It does feel weird to have `OwnerResolver` calculate the owner shares.  Maybe it is just a naming thing?

Reply:
Fixed, move this into `OwnerShares::aggregate`